feat: ACL Manager scripts

sajanrajdev opened the pull request:
Relates to https://github.com/Badger-Finance/badger-multisig/issues/1055

Simple script to find the multisigs where a signer is owner. Additionally, included a couple of scripts to remove members of specific roles from the keeper and registry acls.

gosuto-inzasheru commented:
note that we also have https://github.com/Badger-Finance/badger-multisig/blob/main/scripts/badger/audit_owners.py

gosuto-inzasheru commented:
> @gosuto-inzasheru, decided to add the finder script as it is more specific and agile for this purpose.

as discussed, please see my pr here: https://github.com/Badger-Finance/badger-multisig/pull/1073

sajanrajdev commented:
### Updates

- Removed multisig owner finder script
- Introduced script to audit all ACLs in the infra and generate a report on all the addresses that hold the different ROLES for each. Additionally, allows to identify the ROLES that an specific address has throughout the infra. This script is to be ran for each network separately as it performs RPC calls. Example of run:
```
brownie run scripts/badger/acl_manager.py acl_audit 0xeE8b29AA52dD5fF2559da2C50b1887ADee257556 --network polygon-main-fork
```

gosuto-inzasheru commented on scripts/badger/acl_manager.py:
```diff
@@ -0,0 +1,120 @@
+
+    # build dataframe
+    df = pd.DataFrame(total_data)
+    # Dump result
+    df.to_csv(
+        f"data/badger/acl_roles_audit/acl_roles_audit_{network.show_active()}.csv"
+    )
```
```suggestion
    # Dump result
    os.makedirs("data/badger/acl_roles_audit", exist_ok=True)
    df.to_csv(
        f"data/badger/acl_roles_audit/acl_roles_audit_{network.show_active()}.csv"
    )
```

gosuto-inzasheru commented on scripts/badger/acl_manager.py:
```diff
@@ -0,0 +1,120 @@
+from helpers.addresses import r
```
```suggestion
import os
from helpers.addresses import r
```

gosuto-inzasheru commented on scripts/badger/acl_manager.py:
```diff
@@ -0,0 +1,120 @@
+    df = pd.DataFrame(total_data)
+    # Dump result
+    df.to_csv(
+        f"data/badger/acl_roles_audit/acl_roles_audit_{network.show_active()}.csv"
```
i would add the target address to the filename as well

sajanrajdev replied:
Good call

petrovska-petro reviewed (approved):
nice! v efficient manner to verify roles within the infra ✅ 

```
Target 0xeE8b29AA52dD5fF2559da2C50b1887ADee257556 has the following roles:
+----+-----------------------+-----------------------+-----------------+--------------------------------------------+-------------+
|    | acl_contract          | role                  |   member_number | member_address                             | is_target   |
+====+=======================+=======================+=================+============================================+=============+
| 16 | guardian              | APPROVED_ACCOUNT_ROLE |               3 | 0xeE8b29AA52dD5fF2559da2C50b1887ADee257556 | True        |
+----+-----------------------+-----------------------+-----------------+--------------------------------------------+-------------+
| 27 | keeper                | EARNER_ROLE           |               4 | 0xeE8b29AA52dD5fF2559da2C50b1887ADee257556 | True        |
+----+-----------------------+-----------------------+-----------------+--------------------------------------------+-------------+
| 38 | keeper                | HARVESTER_ROLE        |               3 | 0xeE8b29AA52dD5fF2559da2C50b1887ADee257556 | True        |
+----+-----------------------+-----------------------+-----------------+--------------------------------------------+-------------+
| 49 | keeper                | TENDER_ROLE           |               3 | 0xeE8b29AA52dD5fF2559da2C50b1887ADee257556 | True        |
+----+-----------------------+-----------------------+-----------------+--------------------------------------------+-------------+
| 59 | registryAccessControl | DEVELOPER_ROLE        |               1 | 0xeE8b29AA52dD5fF2559da2C50b1887ADee257556 | True        |
+----+-----------------------+-----------------------+-----------------+--------------------------------------------+-------------+
```
